From 4afd575380d59859baf440af65f42d9248bae8ab Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Sun, 11 Jan 2026 12:09:24 +0100
Subject: [PATCH] Fix cudacodec compilation with HAVE_NVCUVID=HAVE_NVCUVENC=OFF

After the dependencies refactor in https://github.com/opencv/opencv_contrib/commit/d8222d37287f406ba4cd67e578c88c5293ca635c cudacodec no longer builds if HAVE_NVCUVID and HAVE_HVCUVENC are OFF, since the cuda header path is not being added to the include dirs.

Fixes https://github.com/opencv/opencv_contrib/issues/4045

Upstream-Status: Backport [https://github.com/opencv/opencv_contrib/pull/4066]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 modules/cudacodec/CMakeLists.txt | 32 +++++++++++++++-----------------
 1 file changed, 15 insertions(+), 17 deletions(-)

diff --git a/modules/cudacodec/CMakeLists.txt b/modules/cudacodec/CMakeLists.txt
index eeda5714..95a07bdb 100644
--- a/modules/cudacodec/CMakeLists.txt
+++ b/modules/cudacodec/CMakeLists.txt
@@ -30,25 +30,23 @@ if(WITH_NVCUVENC AND NOT HAVE_NVCUVENC)
   message(WARNING "cudacodec::VideoWriter requires Nvidia Video Codec SDK. Please resolve dependency or disable WITH_NVCUVENC=OFF")
 endif()
 
-if(HAVE_NVCUVID OR HAVE_NVCUVENC)
-  if(ENABLE_CUDA_FIRST_CLASS_LANGUAGE)
-      list(APPEND extra_libs CUDA::cuda_driver CUDA::cudart${CUDA_LIB_EXT})
-  else()
-    list(APPEND extra_libs ${CUDA_CUDA_LIBRARY})
-  endif()
-  if(HAVE_NVCUVID)
-    list(APPEND extra_libs ${CUDA_nvcuvid_LIBRARY})
-  endif()
-  if(HAVE_NVCUVENC)
-      if(WIN32)
-        list(APPEND extra_libs ${CUDA_nvencodeapi_LIBRARY})
-      else()
-        list(APPEND extra_libs ${CUDA_nvidia-encode_LIBRARY})
-      endif()
-  endif()
+if(ENABLE_CUDA_FIRST_CLASS_LANGUAGE)
+    list(APPEND extra_libs CUDA::cuda_driver CUDA::cudart${CUDA_LIB_EXT})
+else()
+  list(APPEND extra_libs ${CUDA_CUDA_LIBRARY})
+endif()
+if(HAVE_NVCUVID)
+  list(APPEND extra_libs ${CUDA_nvcuvid_LIBRARY})
+endif()
+if(HAVE_NVCUVENC)
+    if(WIN32)
+      list(APPEND extra_libs ${CUDA_nvencodeapi_LIBRARY})
+    else()
+      list(APPEND extra_libs ${CUDA_nvidia-encode_LIBRARY})
+    endif()
 endif()
 
 ocv_create_module(${extra_libs})
 
 ocv_add_accuracy_tests()
-ocv_add_perf_tests()
\ No newline at end of file
+ocv_add_perf_tests()
-- 
2.43.0

